src/components/main/ShabadBlock.tsx

```diff
@@ -0,0 +1,37 @@
+import {
+  View,
+  Text,
+  SafeAreaView,
+  FlatList,
+} from 'react-native';
+import React, { useState } from 'react';
+import LineBlock from './LineBlock';
+import { EditCtx } from '../../store/context_stores/Contexts';
+import { useValues } from '../../store/StateHooks';
+
+const ShabadBlock = ( { item, currentItems, index } ) => {
+  const { fontSizes, displayElements, sources } = useValues( 'viewerModel' );
+  const { isEditMode, selectedInfo } = EditCtx.useStoreState( ( store ) => ( { ...store } ) );
+  const updatedSelectedInfo = EditCtx.useStoreActions( ( actions ) => actions.updatedSelectedInfo );
+
+  return (
+    <FlatList
+      data={item}
+      keyExtractor={( useless, itemIndex ) => itemIndex.toString()}
+      renderItem={( { index: lineIndex, item: line } ) => (
+        <LineBlock
+          key={`${line.id}-${lineIndex}`}
```

Comment:
from the react docs: `The key only has to be unique among its siblings, not globally unique.` https://reactjs.org/docs/reconciliation.html#keys